### PR TITLE
Enumerable

### PR DESCRIPTION
**Pull request by dannkkie**
Refactgored the code to return enum when there is no block and did a new PR from enumerable branch to development

### CODE REVIEW

File: enumerable.rb [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+
+  # this is the my_map method
+  def my_map(&proc)
+    new_array = []
+    if proc
+      my_each { |i| new_array << proc.call(i) }
+    else
+      my_each { |i| new_array << yield(i) }
+    end
+  end
```
Comment (whiz25) [inline enumerable.rb]: - [ ] should be identical and returns the same thing as ruby's [`map`](https://ruby-doc.org/core-2.6.4/Enumerable.html#method-i-map). ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_enumerable/my_map.png) from [Odin](https://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks#assignment-2))
- [ ] my_map returns a new array with the results of running block once for every element in enum.
- [ ] my_map returns an Enumerator if no block is given
For example:
```ruby
let(:array) { Array.new(100) { rand(0...9) } }
expect(array.my_map(&block)).to eq array.map(&block)
expect(array.my_map).to be_an(Enumerator)
```
You should get `true` if your `my_map` works well.

File: enumerable.rb [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+
+  # this is the my_inject method
+  def my_inject(args = nil)
+    total = args.nil? ? self[0] : args
+    my_each { |i| total = yield(total, i) }
+    total
+  end
```
Comment (whiz25) [inline enumerable.rb]: - [ ] should be identical and returns the same thing as ruby's [`inject`](https://ruby-doc.org/core-2.6.4/Enumerable.html#method-i-inject). ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_enumerable/my_inject.png) from [Odin](https://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks#assignment-2))
- [ ] my_inject combines all elements of enum by applying a binary operation, specified by a block
- [ ] my_inject when a symbol is specified combines each element of the collection by applying the symbol as a named method
For example:
```ruby
let(:range) { Range.new(5, 50) }
actual = range.my_inject(4) { |prod, n| prod * n }
expected = range.inject(4) { |prod, n| prod * n }
expect(actual).to eq expected
```
You should also get `true` if `my_inject` works well.

File: enumerable.rb (hunk 1 above)
Comment (dannkkie) [reply enumerable.rb]: done

File: enumerable.rb (hunk 2 above)
Comment (dannkkie) [reply enumerable.rb]: done

**Comment by dannkkie** [comment]
Thanks

**Pull request merged**